<a href="https://colab.research.google.com/github/LauraDimaria/Project_2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2 - Part 1
###Laura Dimaria
####6/9/22

In [1]:
import pandas as pd

In [3]:
filename = '/content/winequality-white.csv'
df = pd.read_csv(filename, sep = ';')
df.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8        6  
1      9.5        6  
2     10.1        6  
3      9.9        6  
4      9.9        6

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4898 entries, 0 to 4897
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         4898 non-null   float64
 1   volatile acidity      4898 non-null   float64
 2   citric acid           4898 non-null   float64
 3   residual sugar        4898 non-null   float64
 4   chlorides             4898 non-null   float64
 5   free sulfur dioxide   4898 non-null   float64
 6   total sulfur dioxide  4898 non-null   float64
 7   density               4898 non-null   float64
 8   pH                    4898 non-null   float64
 9   sulphates             4898 non-null   float64
 10  alcohol               4898 non-null   float64
 11  quality               4898 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 459.3 KB


1) Source of data
- https://www.kaggle.com/datasets/piyushagni5/white-wine-quality?select=winequality-white.csv

2) Brief description of data
- Dataset is related to chemical components of the Portugese 'Vinho Verde' wine. The wine is historically produced as a white wine so I chose this over the red dataset that was also available. It's also one of my favorite wines to drink.

3) What is the target?
- The target is the quality of the wine, on a scale from 1-10. There are more 'normal' wines than excellent or poor, which would be fun to explore certain features such as risidual sugar and acidity.

4) Is this a classification or regression problem?
- I would probably view this as a classification problem and toy around with the features to see what it would take to create a better wine out of a poor one.

5) How many features?
- 12

6) How many rows of data.
- 4,897

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- The only problem I see is that there is no missing data. The only challange I see myself having is understanding the sulfur dioxide columns, which I can surly read about in one of my somm books.

In [7]:
filename_2 = '/content/energy_efficiency_data.csv'
df_2 = pd.read_csv(filename_2)
df_2.head()

Relative_Compactness  Surface_Area  Wall_Area  Roof_Area  Overall_Height  \
0                  0.98         514.5      294.0     110.25             7.0   
1                  0.98         514.5      294.0     110.25             7.0   
2                  0.98         514.5      294.0     110.25             7.0   
3                  0.98         514.5      294.0     110.25             7.0   
4                  0.90         563.5      318.5     122.50             7.0   

   Orientation  Glazing_Area  Glazing_Area_Distribution  Heating_Load  \
0            2           0.0                          0         15.55   
1            3           0.0                          0         15.55   
2            4           0.0                          0         15.55   
3            5           0.0                          0         15.55   
4            2           0.0                          0         20.84   

   Cooling_Load  
0         21.33  
1         21.33  
2         21.33  
3         21.33  
4         28.28

In [8]:
df_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Relative_Compactness       768 non-null    float64
 1   Surface_Area               768 non-null    float64
 2   Wall_Area                  768 non-null    float64
 3   Roof_Area                  768 non-null    float64
 4   Overall_Height             768 non-null    float64
 5   Orientation                768 non-null    int64  
 6   Glazing_Area               768 non-null    float64
 7   Glazing_Area_Distribution  768 non-null    int64  
 8   Heating_Load               768 non-null    float64
 9   Cooling_Load               768 non-null    float64
dtypes: float64(8), int64(2)
memory usage: 60.1 KB


1) Source of data
  -https://www.kaggle.com/datasets/ujjwalchowdhury/energy-efficiency-data-set

2) Brief description of data
- "We perform energy analysis using 12 different building shapes simulated in Ecotect. The buildings differ with respect to the glazing area, the glazing area distribution, and the orientation, amongst other parameters. We simulate various settings as functions of the afore-mentioned characteristics to obtain 768 building shapes. The dataset comprises 768 samples and 8 features, aiming to predict two real valued responses. It can also be used as a multi-class classification problem if the response is rounded to the nearest integer."

3) What is the target?
- Heating or Cooling Load, I would probably choose cooling load because of global warming.

4) Is this a classification or regression problem?
- Regression to predict the energy effeciency of a building

5) How many features?
- 10

6) How many rows of data.
- 768

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- While I am all about energy effeciency, some of the feature terms are foreign to me and I would have to learn more. There is also not a whole lot of data which could lend to some not high functioning models.